src/Albany_StateManager.cpp

```diff
@@ -428,22 +428,204 @@ Albany::StateManager::registerSideSetStateVariable(
+  {
+    mfe_type = StateStruct::NodalData;                  // One value per node
+  } else if (dl->rank() == 1 && dl->name(1) == "Side")  // Element data
+  {
+    mfe_type = StateStruct::ElemData;  // One value per element
+  } else if (dl->rank() > 1) {
+    if (dl->name(1) == "Dim")
+      mfe_type = StateStruct::ElemData;   // One vector/tensor per element
+    else if (dl->name(1) == "Node")       // Element node data
+      mfe_type = StateStruct::ElemNode;   // One value per side node
+    else if (dl->name(1) == "QuadPoint")  // Quad point data
+      mfe_type = StateStruct::QuadPoint;  // One value per side quad point
+    else
+      TEUCHOS_TEST_FOR_EXCEPTION(
+          true, std::logic_error, "StateManager: Unknown Entity type.\n");
+  } else {
+    TEUCHOS_TEST_FOR_EXCEPTION(
+        true, std::logic_error, "StateManager: Unknown Entity type.\n");
+  }
+
+  sis_ptr->push_back(Teuchos::rcp(new StateStruct(stateName, mfe_type)));
+  StateStruct& stateRef = *sis_ptr->back();
+  stateRef.setInitType(init_type);
+  stateRef.setInitValue(init_val);
+  stateRef.setMeshPart(meshPartName);
+
+  if (stateRef.entity == StateStruct::NodalData) {
+    TEUCHOS_TEST_FOR_EXCEPTION(
+        dl->name(0) == "Node",
+        std::logic_error,
+        "Error! NodalData states should have dl <Node,...>.\n");
+
+    dl->dimensions(stateRef.dim);
+
+    // Register the state with the nodalDataVector also.
+    Teuchos::RCP<Adapt::NodalDataBase> nodalDataBase =
+        getSideSetNodalDataBase(sideSetName);
+
+    if (dl->rank() == 1)  // node scalar
+      nodalDataBase->registerVectorState(stateName, 1);
+    else if (dl->rank() == 2)  // node vector
+      nodalDataBase->registerVectorState(stateName, stateRef.dim[1]);
+    else if (dl->rank() == 3)  // node tensor
+      nodalDataBase->registerVectorState(
+          stateName, stateRef.dim[1] * stateRef.dim[2]);
+  } else if (
+      stateRef.entity == StateStruct::NodalDataToElemNode ||
+      stateRef.entity == Albany::StateStruct::NodalDistParameter) {
+    // Strip one dimension out cause it's Side
+    TEUCHOS_TEST_FOR_EXCEPTION(
+        dl->rank() < 2,
+        std::logic_error,
+        "Error! The given layout does not appear to be that of a side set "
+        "field.\n");
+
+    dl->dimensions(stateRef.dim);
+
+    // Register the state with the nodalDataVector also.
+    Teuchos::RCP<Adapt::NodalDataBase> nodalDataBase =
+        getSideSetNodalDataBase(sideSetName);
+
+    // These entities store data as <Cell,Side,Node,...>, so the dl has one more
+    // rank
+    if (dl->rank() == 2)  // node scalar
+      nodalDataBase->registerVectorState(stateName, 1);
+    else if (dl->rank() == 3)  // node vector
+      nodalDataBase->registerVectorState(stateName, stateRef.dim[1]);
+    else if (dl->rank() == 4)  // node tensor
+      nodalDataBase->registerVectorState(
+          stateName, stateRef.dim[1] * stateRef.dim[2]);
+  } else {
+    // Strip one dimension out cause it's Side
```

Comment:
Max, isn't this impossible? I mean, if you use `registerSideSetState_collapsed`, you should *not* have the `Side` tag, no?

Reply:
Yep, the comment just got left in but no dimension gets stripped in registerSideSetStateVariable_collapsed. I'll update the comments there, thanks.

Reply:
Rather than simply update the comment, I would replace the whole body of the else statement with an exception. If someone calles the `_collapsed` version with the wrong layout, we should throw.